### PR TITLE
Rework SLO metrics

### PR DESCRIPTION
**Pull request by joe-elliott**
**What this PR does**:
This PR reworks the way our SLO metrics are recorded with these goals in mind;

- Push the calculations out to be as close the http.Server as possible
- Clearly metric both total queries and SLO queries in one spot
- Cleanup and remove unused metrics

This PR accomplishes the above with the following sharp corners:

- In order to pass the throughput up to the code that calculates the SLOs a *float was added to the context. This works, but is brittle and difficult to detect if it breaks.
- Specialized code needed to be added to the streaming endpoint

TODO
- [x] correct changelog
- [x] add tests

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: modules/frontend/slos.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	"net/http"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+const (
+	traceByIDOp = "traces"
+	searchOp    = "search"
```
Comment (zalegrala) [inline modules/frontend/slos.go:14]: Do we want to track SLOs on the other frontend operation?  Looks like we lost a couple.

**Review by zalegrala — COMMENTED** [review]
This looks reasonable to me.  Just a question about tracking the other operations.  Have you had a chance to test this out?

File: modules/frontend/slos.go (hunk 1 above)
Comment (joe-elliott) [reply modules/frontend/slos.go:14]: We currently don't support "policy based" SLOs for those other ops and just use `tempo_request_duration_seconds` histograms. Only these two operations were using these metrics for SLOs so I removed the other two to align `tempo_query_frontend_queries_total` and `tempo_query_frontend_queries_within_slo_total`. Previously, `tempo_query_frontend_queries_total` had more values for the `op` label then `tempo_query_frontend_queries_within_slo_total` and I wanted this pair of metrics to be aligned.

**Review by mdisibio — APPROVED** [review]
LGTM, accuracy of the metrics for SLO calculation is important and I think this is a good change.  I think this is Tempo's first case of context smuggling so it gets a pass. 😄

**Review by zalegrala — APPROVED** [review]
Nice work Joe.  Thanks for giving this area some :heart: .

**Pull request merged**